File: DuDoong-Domain/src/main/java/band/gosrock/domain/common/events/Events.java
package band.gosrock.domain.common.events;


import org.springframework.context.ApplicationEventPublisher;

public class Events {
    private static ThreadLocal<ApplicationEventPublisher> publisherLocal = new ThreadLocal<>();

    public static void raise(DomainEvent event) {
        if (event == null) return;

        if (publisherLocal.get() != null) {
            publisherLocal.get().publishEvent(event);
        }
    }

    public static void setPublisher(ApplicationEventPublisher publisher) {
        publisherLocal.set(publisher);
    }

    public static void reset() {
        publisherLocal.remove();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/common/aop/domainEvent/Events.kt
package band.gosrock.domain.common.aop.domainEvent

import org.springframework.context.ApplicationEventPublisher

object Events {
    private val publisherLocal = ThreadLocal<ApplicationEventPublisher>()

    @JvmStatic
    fun raise(event: DomainEvent?) {
        if (event == null) return
        publisherLocal.get()?.publishEvent(event)
    }

    @JvmStatic
    fun setPublisher(publisher: ApplicationEventPublisher) {
        publisherLocal.set(publisher)
    }

    @JvmStatic
    fun reset() {
        publisherLocal.remove()
    }
}
